swift/swift.py

```diff
@@ -204,50 +227,58 @@ def _swiftcall(self, msg: str):
               "destory": Destroy an app.
                 Its "args" is a dictionary with a key; "name".
                 The value of "name" is a name of app you want to destroy.
+        
+        Raises:
+            RuntimeError: When the user rejects the request.
+        
+        Returns:
+            The returned value of the swift-call, if any.
         """
-        try:
-            msg = json.loads(msg)
-        except json.JSONDecodeError as e:
-            print(f"swift.swift._swiftcall(): {e!r}")
-            return
-        if any(key not in msg for key in ("action", "args")):
-            print("The message was ignored because "
-                  "it has no such key; action or args.")
-            return
+        msg = json.loads(msg)

```

Comment:
Is it okay not to check if `msg` has a correct form?

Reply:
I wrote

> This can raise an exception if the arguments do not follow the valid API.
The caller must obey the API and catch the possible exceptions.

in the docstring of `_handleSwiftcall()`.

I understand that we agreed to catch all the exceptions as possible to prevent the application to crash.
However, this is only called by `_swiftcall()` and it catches all the exceptions.
Moreover, this is an internal method, i.e., will not be used in other places.
I think this usage is safe enough!

The main reason of not catching the exceptions is that the exception information should be transferred to the caller app.